x-pack/plugins/fleet/server/services/epm/packages/bulk_install_packages.ts

```diff
@@ -42,29 +45,55 @@ export async function bulkInstallPackages({
   const bulkInstallResults = await Promise.allSettled(
     packagesResults.map(async (result, index) => {
       const packageName = getNameFromPackagesToInstall(packagesToInstall, index);
+
       if (result.status === 'fulfilled') {
         const pkgKeyProps = result.value;
         const installedPackageResult = await isPackageVersionOrLaterInstalled({
           savedObjectsClient,
           pkgName: pkgKeyProps.name,
           pkgVersion: pkgKeyProps.version,
         });
+
         if (installedPackageResult) {
           const {
             name,
             version,
             installed_es: installedEs,
             installed_kibana: installedKibana,
           } = installedPackageResult.package;
-          return {
-            name,
-            version,
-            result: {
-              assets: [...installedEs, ...installedKibana],
-              status: 'already_installed',
-              installType: installedPackageResult.installType,
-            } as InstallResult,
-          };
+
+          const isManagedPackage = [...DEFAULT_PACKAGES, ...AUTO_UPDATE_PACKAGES].some(
```

Comment:
I think we need this additional logic to support what we're trying to do in https://github.com/elastic/kibana/issues/111858

What we want to be true, in order to support "stack-aligned" packages properly is the following:

- When Kibana boots up (and Fleet setup runs), Fleet attempts to install preconfigured packages _and_ checks the installation status of any package that appears in `AUTO_UPDATE`
- If a version of a `default` or `auto update` package is installed that is _not_ the current `latest` version of the package, it should be uninstalled and replaced with the `latest` version of the package

This means that if a rollback occurs from 8.0 -> 7.16, we should uninstall the version of APM or System that's valid for Kibana 8.0 and replace it with the version that's valid for 7.16.

This chunk of logic in the diff here is definitely not covering all this, but it's just a rough guess/first pass for now. Would love some input on this and any context that would help narrow the changes we have to make.

Reply:
I think the behavior you have here generally makes sense. I do wonder if we should hoist this logic up a level to be part of `setup` instead of `bulkInstallPackages` to make testing `bulkInstallPackages` independent of setup logic easier and make the regular `installPackage` and `bulkInstallPackages` consistent in behavior. Any danger to removing the package from the setup level that I'm not considering?

Reply:
Yeah you are definitely right - this should be hoisted up rather than being tied to this `bulkInstall` operation. I think there's some slight re-querying going on if we take that path, but nothing substantial enough to warrant performance concerns for me.

Reply:
Moved this to an `ensureManagedPackageVersion` method in our `setup` service in https://github.com/elastic/kibana/pull/118797/commits/a788336f30a352e14703127efef788986bb2e719